Add docs & api-docs for Azure Secrets Static Roles (#908)
hashicorp/web-unified-docs: content/vault/v1.21.x (rc)/content/docs/secrets/azure.mdx

Please see the [API documentation](/vault/api-docs/secret/azure#configure-access)
for more details on the fields associated with plugin WIF.

> ## Static Roles

Reviewer: ```suggestion
## Static roles
```
Style correction: use sentence case for headings

@include 'alerts/enterprise-only.mdx'

> In addition to dynamic roles, the Azure secrets engine also supports static roles.

Reviewer: ```suggestion
The Azure secrets engine supports dynamic and static roles.
```
Style correction: write simply with direct language

> A static role allows Vault to manage a long-lived client secret for an Azure application.
> Unlike dynamic roles, which generate new credentials on each request, a static role always returns
> the same client secret until it is rotated. Rotation can be triggered manually by calling the
> [`rotate-role`](/vault/api-docs/secret/azure#manually-rotate-static-role-password) endpoint. When it occurs, Vault
> will generate a new client secret for the application and revoke the previous one.
> 
> Please see [API documentation](/vault/api-docs/secret/azure#create-update-static-role) for more details on this feature

Reviewer: ```suggestion
Vault uses static roles to manage long-lived client secrets for Azure
application. Unlike dynamic roles, which generate new credentials on each
request, static roles always return the same client secret until you explicitly
rotate the value. Once the secret rotates, Vault generates a new client secret
for the application and revokes the previous secret.

You can manually rotate secrets by calling the
[`rotate-role`](/vault/api-docs/secret/azure#manually-rotate-static-role-password)
endpoint. Refer to the
[Azure plugin API documentation](/vault/api-docs/secret/azure#create-update-static-role)
for more details.
```
Style correction: write in active voice, use meaningful anchor text
